test/e2e/mock-metalsmith-custom/meta.js

```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  "metalsmith": function (metalsmith, opts, helpers) {
+    function customMetalsmithPlugin (files, metalsmith, done) {
+      // Implement something really custom here.
+
+      var readme = files['readme.md']
+      delete files['readme.md']
+      files['custom/readme.md'] = readme
```

Comment:
Just to be sure. What you're doing here in custom metalsmith plugin is you're moving file from root template directory into `custom` directory, right?